crates/telio-wg/src/adapter/boring.rs

```diff
@@ -36,7 +36,18 @@ impl BoringTun {
         firewall_reset_connections_callback: super::FirewallResetConnsCb,
     ) -> Result<Self, AdapterError> {
         let config = DeviceConfig {
-            n_threads: 4,
+            // Apple's Boringtun device runs most efficiently on a single perf-core
+            n_threads: {
+                if cfg!(not(any(
+                    target_os = "ios",
+                    target_os = "macos",
+                    target_os = "tvos"
+                ))) {
+                    num_cpus::get()
```

Comment:
Are you sure that we should take all of the cpus available? Maybe the overhead is not worth it? Maybe this `4` which is used also by `boringtun-cli` was the right choice?

Reply:
I think it is worth to try ... I think, we'll have results from Android and Linux team the same day as release. If it will result in poor performance, it will be a literally one-liner fix, to revert it :smiley_cat:

Reply:
Ok, let's merge it as it is